### PR TITLE
Feat(eos_designs): Add `ptp_settings` to replace `ptp` key in eos_designs

### LINKED ISSUE #3236: eos_designs `ptp` key should be renamed to `ptp_settings` to avoid conflicting variable with eos_cli_config_gen
- Implement new `ptp_settings:` key in eos_designs
- deprecate `ptp:` key in eos_designs and key to be remove in AVD 5.0.0
  - Add note that if `ptp_settings` is defined it would take precedence.
- The `ptp` schema would use a $ref to ptp_settings to avoid maintaining two models.

### PR DESCRIPTION
**Pull request by laxmikantchintakindi**
## Change Summary

 Rename ptp  key to  ptp_settings to avoid conflicting variable with  eos_cli_config_gen.

## Related Issue(s)

Fixes #3236

## Component(s) name

`arista.avd.eos_designs`

### User Checklist

- N/A

### Repository Checklist

- [x] My code has been rebased from devel before I start
- [x] I have read the [**CONTRIBUTING**](https://avd.sh/en/latest/docs/contribution/overview.html) document.
- [x] My change requires a change to the documentation and documentation have been updated accordingly.
- [x] I have updated [molecule CI](https://github.com/aristanetworks/avd/tree/devel/ansible_collections/arista/avd/molecule) testing accordingly. (check the box if not applicable)

### CODE REVIEW

File: python-avd/pyavd/_eos_designs/schema/eos_designs.schema.yml [hunk 1]
```diff
@@ -3180,25 +3180,14 @@ keys:
 
             '
   ptp:
+    deprecation:
```
Comment (Shivani-gslab) [inline python-avd/pyavd/_eos_designs/schema/eos_designs.schema.yml:3186]: please add a test for ptp in `eos_designs_unit_tests_deprecated_vars` moelcule

File: python-avd/pyavd/_eos_designs/schema/schema_fragments/ptp_settings.schema.yml [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+# yaml-language-server: $schema=../../../../plugins/plugin_utils/schema/avd_meta_schema.json
+# Line above is used by RedHat's YAML Schema vscode extension
+# Use Ctrl + Space to get suggestions for every field. Autocomplete will pop up after typing 2 letters.
+type: dict
+keys:
+  ptp_settings:
+    documentation_options:
+      table: ptp_settings
+    type: dict
+    description: The `ptp` setting is deprecated and it will be removed in version 5.0.0 to avoid conflict with `ptp` setting in eos_cli_config_gen. `ptp_settings` takes precedence when both `ptp` and `ptp_settings` are set.
```
Comment (MaheshGSLAB) [inline python-avd/pyavd/_eos_designs/schema/schema_fragments/ptp_settings.schema.yml]: need better description

File: ansible_collections/arista/avd/roles/eos_designs/docs/tables/ptp.md [hunk 3]
```diff
@@ -25,6 +25,10 @@
 === "YAML"
 
     ```yaml
+    # The `ptp` setting is deprecated and it will be removed in version 5.0.0 to avoid conflict with `ptp` setting in eos_cli_config_gen. `ptp_settings` takes precedence when both `ptp` and `ptp_settings` are set.
```
Comment (MaheshGSLAB) [inline ansible_collections/arista/avd/roles/eos_designs/docs/tables/ptp.md]: should we move ptp_profiles documentation also in ptp_settings.md
@ClausHolbechArista 

[line number 37](https://github.com/aristanetworks/avd/pull/4155/files#diff-be639d3b605ad40103adc39c4cccb48fc9d17a81e66a4b9bc4b84c6067399b00L33)

File: python-avd/pyavd/_eos_designs/schema/schema_fragments/ptp_settings.schema.yml (hunk 2 above)
Comment (ClausHolbechArista) [inline python-avd/pyavd/_eos_designs/schema/schema_fragments/ptp_settings.schema.yml]:
```suggestion
    # TODO AVD5.0: Update description to not mention ptp key.
    description: |-
      Common PTP settings.
      `ptp_settings` replaces the old `ptp` key. `ptp_settings` takes precedence.
```

**Review by ClausHolbechArista — CHANGES_REQUESTED** [review]

File: python-avd/pyavd/_eos_designs/schema/eos_designs.schema.yml (hunk 1 above)
Comment (laxmikantchintakindi) [reply python-avd/pyavd/_eos_designs/schema/eos_designs.schema.yml:3186]: Added.

File: python-avd/pyavd/_eos_designs/schema/schema_fragments/ptp_settings.schema.yml (hunk 2 above)
Comment (laxmikantchintakindi) [reply python-avd/pyavd/_eos_designs/schema/schema_fragments/ptp_settings.schema.yml]: Fixed.

Comment (laxmikantchintakindi) [reply python-avd/pyavd/_eos_designs/schema/schema_fragments/ptp_settings.schema.yml]: Fixed.

File: ansible_collections/arista/avd/roles/eos_designs/docs/tables/ptp.md (hunk 3 above)
Comment (laxmikantchintakindi) [reply ansible_collections/arista/avd/roles/eos_designs/docs/tables/ptp.md]: Moved.

**Review by Vibhu-gslab — DISMISSED** [review]
LGTM!

**Review by MaheshGSLAB — DISMISSED** [review]

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/docs/input-variables.md [hunk 4]
```diff
@@ -1043,10 +1043,10 @@ roles/eos_cli_config_gen/docs/tables/platform.md
 roles/eos_cli_config_gen/docs/tables/poe.md
 --8<--
 
-### PTP
+### PTP settings
```
Comment (carlbuchmann) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/docs/input-variables.md]: Please revert this change, and update the eos_designs `input-variables.md` documentation.
Additionally, I beleive we would want to keep both `ptp` and `ptp_settings`, so that users have the deprecation documentation.

Comment (laxmikantchintakindi) [reply ansible_collections/arista/avd/roles/eos_cli_config_gen/docs/input-variables.md]: Reverted this and updated eos_designs input-variables.md.

File: ansible_collections/arista/avd/roles/eos_designs/docs/tables/ptp.md [hunk 5]
```diff
@@ -7,72 +7,23 @@
...
-    | [<samp>&nbsp;&nbsp;&nbsp;&nbsp;transport</samp>](## "ptp_profiles.[].transport") | String |  |  | Valid Values:<br>- <code>ipv4</code> |  |
 
 === "YAML"
 
     ```yaml
+    # Common PTP settings.
+    # `ptp_settings` replaces the old `ptp` key. `ptp_settings` takes precedence.
+    # This key is deprecated.
+    # Support will be removed in AVD version v5.0.0.
+    # Use <samp>ptp_settings</samp> instead.
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_designs/docs/tables/ptp.md]: This looks double. Please remove the change to description on the old key, since the deprecation stuff will already take care of it.

File: ansible_collections/arista/avd/roles/eos_designs/docs/input-variables.md [hunk 6]
```diff
@@ -1350,14 +1350,22 @@ Management interface is modified for specific platforms like modular platforms w
...
 --8<--
 
+## PTP settings
+
+See the [Configuring PTP](how-to/ptp_settings.md) how-to for details.
+
+--8<--
+roles/eos_designs/docs/tables/ptp_settings.md
+--8<--
+
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_designs/docs/input-variables.md]: I think a single section is enough here and then combining all the keys in the single table.

```suggestion
## PTP settings

See the [Configuring PTP](how-to/ptp.md) how-to for details.

--8<--
roles/eos_designs/docs/tables/ptp_settings.md
--8<--

```

File: python-avd/pyavd/_eos_designs/schema/schema_fragments/ptp.schema.yml [hunk 7]
```diff
@@ -7,22 +7,11 @@
 type: dict
 keys:
   ptp:
+    deprecation:
+      warning: true
+      new_key: 'ptp_settings'
+      remove_in_version: v5.0.0
     documentation_options:
       table: ptp
```
Comment (ClausHolbechArista) [inline python-avd/pyavd/_eos_designs/schema/schema_fragments/ptp.schema.yml]:
```suggestion
      table: ptp_settings
```

File: python-avd/pyavd/_eos_designs/structured_config/base/__init__.py [hunk 8]
```diff
@@ -663,8 +663,7 @@ def ptp(self) -> dict | None:
             if get(self._hostvars, "ptp") is not None:
```
Comment (ClausHolbechArista) [inline python-avd/pyavd/_eos_designs/structured_config/base/__init__.py:664]:
```suggestion
            # TODO: AVD5.0.0 Remove this handling since the `ptp` key is removed from eos_designs.
            if get(self._hostvars, "ptp") is not None:
```

File: ansible_collections/arista/avd/roles/eos_designs/docs/how-to/ptp_settings.md
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_designs/docs/how-to/ptp_settings.md:1]: Instead of creating another file, please just update the existing one.

**Review by ClausHolbechArista — CHANGES_REQUESTED** [review]

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/docs/input-variables.md (hunk 4 above)
Comment (ClausHolbechArista) [reply ansible_collections/arista/avd/roles/eos_cli_config_gen/docs/input-variables.md]: I am suggesting the opposite. A single documentation section, with a single table, but that table contains both the new and old key. That is how we do elsewhere.

Comment (laxmikantchintakindi) [reply ansible_collections/arista/avd/roles/eos_cli_config_gen/docs/input-variables.md]: I added documentation of both new and old key in a single markdown file. Please review again.

File: ansible_collections/arista/avd/roles/eos_designs/docs/tables/ptp.md (hunk 5 above)
Comment (laxmikantchintakindi) [reply ansible_collections/arista/avd/roles/eos_designs/docs/tables/ptp.md]: Fixed.

File: python-avd/pyavd/_eos_designs/schema/schema_fragments/ptp.schema.yml (hunk 7 above)
Comment (laxmikantchintakindi) [reply python-avd/pyavd/_eos_designs/schema/schema_fragments/ptp.schema.yml]: Fixed.

File: ansible_collections/arista/avd/roles/eos_designs/docs/how-to/ptp_settings.md
Comment (laxmikantchintakindi) [reply ansible_collections/arista/avd/roles/eos_designs/docs/how-to/ptp_settings.md:1]: Rename ptp.md to ptp_settings.md.

**Comment by Vibhu-gslab** [comment]
Regarding CI failure: we need to change `ptp.md` in https://github.com/aristanetworks/avd/blob/devel/mkdocs.yml#L152

File: ansible_collections/arista/avd/molecule/eos_designs_deprecated_vars/inventory/host_vars/host1/ptp.yml [hunk 9]
```diff
@@ -0,0 +1,16 @@
...
+l3leaf:
+  defaults:
+    loopback_ipv4_pool: 192.168.255.0/24
+    vtep_loopback_ipv4_pool: 192.168.254.0/24
+    id: 101
+    bgp_as: 101
+  nodes:
+    host1:
+      id: 101
+      bgp_as: 101
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/molecule/eos_designs_deprecated_vars/inventory/host_vars/host1/ptp.yml]: Since this is a single shared host in the deprecated vars, we should not set all these common things here. Just the ptp key.

File: ansible_collections/arista/avd/roles/eos_designs/docs/how-to/ptp_settings.md [hunk 10]
```diff
@@ -183,6 +190,13 @@ If you prefer to have PTP clock identity be the system MAC-address of the switch
 
 - on the fabric level, for example FABRIC.yml
 
+  ```yaml
+  ptp_settings:
+    auto_clock_identity: < true | false | default -> true >
+  ```
+
+  OR
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_designs/docs/how-to/ptp_settings.md]: Please just remote all mentions of the old key here. We do not have to advice how to use a model that we have deprecated.

Comment (ClausHolbechArista) [reply ansible_collections/arista/avd/roles/eos_designs/docs/how-to/ptp_settings.md]: someone who really wants to know can still see the `ptp` variant in the input variables tables.

File: ansible_collections/arista/avd/roles/eos_designs/docs/how-to/ptp_settings.md
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_designs/docs/how-to/ptp_settings.md:1]: Please revert this rename. It is not only describing one key like ptp_settings, but the whole subject of PTP, so the old name `ptp.md` is more correct.

File: ansible_collections/arista/avd/roles/eos_designs/docs/input-variables.md [hunk 11]
```diff
@@ -1352,10 +1352,10 @@ roles/eos_designs/docs/tables/platform-settings.md
 
 ## PTP settings
 
-See the [Configuring PTP](how-to/ptp.md) how-to for details.
+See the [Configuring PTP](how-to/ptp_settings.md) how-to for details.
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_designs/docs/input-variables.md]:
```suggestion
See the [Configuring PTP](how-to/ptp.md) how-to for details.
```

**Review by ClausHolbechArista — CHANGES_REQUESTED** [review]

File: ansible_collections/arista/avd/molecule/eos_designs_deprecated_vars/inventory/host_vars/host1/ptp.yml (hunk 9 above)
Comment (laxmikantchintakindi) [reply ansible_collections/arista/avd/molecule/eos_designs_deprecated_vars/inventory/host_vars/host1/ptp.yml]: Kept only ptp key and removed remaining settings.

File: ansible_collections/arista/avd/roles/eos_designs/docs/how-to/ptp_settings.md (hunk 10 above)
Comment (laxmikantchintakindi) [reply ansible_collections/arista/avd/roles/eos_designs/docs/how-to/ptp_settings.md]: Fixed. But we are not guiding how to set`ptp` in input-variables.md. He can refer roles/eos_designs/docs/tables/ptp_settings.md right?

File: ansible_collections/arista/avd/roles/eos_designs/docs/how-to/ptp_settings.md
Comment (laxmikantchintakindi) [reply ansible_collections/arista/avd/roles/eos_designs/docs/how-to/ptp_settings.md:1]: Fixed.

File: ansible_collections/arista/avd/roles/eos_designs/docs/how-to/ptp_settings.md (hunk 10 above)
Comment (ClausHolbechArista) [reply ansible_collections/arista/avd/roles/eos_designs/docs/how-to/ptp_settings.md]: Exactly. The table contains the old and new as a reference, but the "guide" should only talk about the new stuff.

**Review by ClausHolbechArista — APPROVED** [review]
LGTM! Thank you.

**Review by MaheshGSLAB — APPROVED** [review]
LGTM

**Review by gmuloc — APPROVED** [review]
